### DISCUSSION TITLE
Getting `remote-exec provisioner error` running `terraform apply -auto-approve`

### DISCUSSION THREAD

**Discussion by TimHeckel** [kube-hetzner/terraform-hcloud-kube-hetzner#268] [Q&A] [answered]
Not sure if this is on hetzner's side or in my terraform config. The command worked before, so I think this is a strange hetzner thing. This error repeats 6 times - for each node in my cluster (3 agent nodes, 3 control plane nodes):

```
│ Error: remote-exec provisioner error
│
│   with module.kube-hetzner.module.control_planes["0-2-sb-control-plane"].hcloud_server.server,
│   on .terraform/modules/kube-hetzner/modules/host/main.tf line 61, in resource "hcloud_server" "server":
│   61:   provisioner "remote-exec" {
│
│ Error connecting to SSH_AUTH_SOCK: dial unix /private/tmp/com.apple.launchd.2j15SMcoua/Listeners: connect: no such file or directory
```

**Comment by phaer** [comment, marked as answer]
This looks like you've got an SSH_AUTH_SOCK set in your shell, but it's referring to a socket with does not exist on your local macOS system.
This could be because there's actually no SSH agent running on your system. I don't use macOS, so I can't help you here.

If you are intending to simply use an SSH key stored in a file which you declared in you terraform vars, you could try to
`unset SSH_AUTH_SOCK` in you shell before running `terraform apply`, then `ssh` shouldn't look for a socket there.

EDIT: If my guess is correct, you should get a similar error if you try to ssh to any other host (might depend on details in your local ssh config).

**Reply by TimHeckel** [reply]
@phaer  - thanks so much. That was it. I honestly have no idea how my machine got into that state, I was switching between a rosetta terminal session and an m1 terminal session, but...who knows. I restarted and all is well. Thanks again! I wasn't sure if the error was originating on the host machine or my machine.
